examples/training/classification/imagenet/basic_training.py

```diff
@@ -142,9 +145,16 @@ def augment(img, label):
 Different distribution strategies may be used for different training hardware, as indicated below.
 """
 
-# For TPU training, use tf.distribute.TPUStrategy()
-# MirroredStrategy is best for a single machine with multiple GPUs
-strategy = tf.distribute.MirroredStrategy()
+
```

Comment:
2 comments:

1) you are using old code for instantiating a TPUStrategy. There is a shorter version now: TPUClusterResolver.connect(). See example here: https://bit.ly/keras-TPU

2) it is always more user-friendly to have the code detect the hardware automatically (instead of a flag). This is easy to do with distribution strategies. Also see: https://bit.ly/keras-TPU